src/algebra/module/linear_map.lean

```diff
@@ -632,6 +632,9 @@ instance : module S (M →ₗ[R] M₂) :=
 { add_smul := λ a b f, ext $ λ x, add_smul _ _ _,
   zero_smul := λ f, ext $ λ x, zero_smul _ _ }
 
+instance [no_zero_smul_divisors S M₂] : no_zero_smul_divisors S (M →ₗ[R] M₂) :=
+coe_injective.no_zero_smul_divisors _ rfl (λ _ _, funext $ λ _, rfl)
```

Comment:
`λ _ _, funext $ λ _, rfl` should be its own lemma, `coe_smul`

Reply:
(note this isn't about making the proof as short as possible, it's about spotting all the subproofs that I expect we'll want independently)

Reply:
I've added those lemmas (untested, will need to see if it builds OK in CI). For `alternating_map` I called it `coe_fn_smul` since `alternating_map.coe_fn_smul` already exists (about the coercion to `multilinear_map`).